mod/discussions/views/default/forms/discussion/save.php

```diff
@@ -12,58 +12,70 @@
 $container_guid = elgg_extract('container_guid', $vars);
 $guid = elgg_extract('guid', $vars, null);
 
-?>
-<div>
-	<label><?php echo elgg_echo('title'); ?></label><br />
-	<?php echo elgg_view('input/text', array('name' => 'title', 'value' => $title)); ?>
-</div>
-<div>
-	<label><?php echo elgg_echo('discussion:topic:description'); ?></label>
-	<?php echo elgg_view('input/longtext', array('name' => 'description', 'value' => $desc)); ?>
-</div>
-<div>
-	<label><?php echo elgg_echo('tags'); ?></label>
-	<?php echo elgg_view('input/tags', array('name' => 'tags', 'value' => $tags)); ?>
-</div>
-<div>
-    <label><?php echo elgg_echo("discussion:topic:status"); ?></label><br />
-	<?php
-		echo elgg_view('input/select', array(
-			'name' => 'status',
-			'value' => $status,
-			'options_values' => array(
-				'open' => elgg_echo('status:open'),
-				'closed' => elgg_echo('status:closed'),
-			),
-		));
-	?>
-</div>
-<div>
-	<label><?php echo elgg_echo('access'); ?></label><br />
-	<?php echo elgg_view('input/access', array(
+$fields = [
+	[
+		'type' => 'text',
+		'name' => 'title',
+		'value' => $title,
+		'label' => elgg_echo('title'),
+		'required' => true,
+	],
+	[
+		'type' => 'longtext',
+		'name' => 'description',
+		'value' => $desc,
+		'label' => elgg_echo('discussion:topic:description'),
+		'required' => true,
+	],
+	[
+		'type' => 'tags',
+		'name' => 'tags',
+		'value' => $tags,
+		'label' => elgg_echo('tags'),
+	],
+	[
+		'type' => 'select',
+		'name' => 'status',
+		'value' => $status,
+		'options_values' => array(
+			'open' => elgg_echo('status:open'),
+			'closed' => elgg_echo('status:closed'),
+		),
+		'label' => elgg_echo('discussion:topic:status'),
+	],
+	[
+		'type' => 'access',
 		'name' => 'access_id',
 		'value' => $access_id,
 		'entity' => get_entity($guid),
 		'entity_type' => 'object',
 		'entity_subtype' => 'discussion',
-	)); ?>
-</div>
-<div class="elgg-foot">
-<?php
+		'label' => elgg_echo('access'),
+	],
+];
+
+$fields = elgg_trigger_plugin_hook('fields', 'forms/discussion/save', null, $fields);
 
-echo elgg_view('input/hidden', array(
+$fields[] = [
```

Comment:
I'd rather hand the hook handlers an object that can take care of that stuff. E.g. something like [MenuList](https://gist.github.com/mrclay/2242916), which makes inserting/reordering easy.